internal/controller/controllers/agent_controller.go

```diff
@@ -822,6 +825,41 @@ func (r *AgentReconciler) updateInventory(log logrus.FieldLogger, host *models.H
 	return nil
 }
 
+func updateInventoryLabels(agent *aiv1beta1.Agent) {
+	inventory := agent.Status.Inventory
+	if agent.ObjectMeta.Labels == nil {
+		agent.ObjectMeta.Labels = make(map[string]string)
+	} else {
+		for key := range agent.ObjectMeta.Labels {
```

Comment:
what is the performance of this with a lot of hosts? 
Doing this operation on every reconcile sounds expensive. 
Not that i have an alternative, just want to know what will be the affect

Reply:
Good question.  I assumed at this point that we would have just a few labels, and probably cheaper than updating inventory on every reconcile.  This method should only run when updateInventory() runs, and we should optimize when updateInventory() runs because in most cases, once would be enough.

Reply:
update inventory is called on every reconcile.
It's not optimal as well.

Reply:
Exactly.  We should optimize that, then this will be optimized as well.

Reply:
Optimized in latest PR